From 8318c99cceb629a7a4f5bc4ff6bd78972e83ecb6 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Tue, 20 Oct 2015 17:55:00 +0300
Subject: [PATCH 04/51] jni/libfm_jni: fix calibration data path for legacy
 fm_qsoc_patches

* Set 'TARGET_FM_LEGACY_PATCHLOADER := true' to enable.

Change-Id: Ibfa83ba549b36ad5f93bacf310459bdb0ce33987
---
 jni/Android.mk       | 4 ++++
 jni/FmConst.h        | 4 ++++
 libfm_jni/Android.mk | 4 ++++
 libfm_jni/FM_Const.h | 4 ++++
 4 files changed, 16 insertions(+)

diff --git a/jni/Android.mk b/jni/Android.mk
index bf12d79..b536356 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -24,6 +24,10 @@ LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
 LOCAL_MODULE := libqcomfm_jni
 LOCAL_MODULE_TAGS := optional
 
+ifeq ($(TARGET_FM_LEGACY_PATCHLOADER),true)
+    LOCAL_CFLAGS += -DFM_LEGACY_PATCHLOADER
+endif
+
 include $(BUILD_SHARED_LIBRARY)
 
 
diff --git a/jni/FmConst.h b/jni/FmConst.h
index e37160f..f503f83 100644
--- a/jni/FmConst.h
+++ b/jni/FmConst.h
@@ -44,7 +44,11 @@ const UINT CAL_DATA_SIZE = 23;
 #define STD_BUF_SIZE  256
 
 const char *const FM_PERFORMANCE_PARAMS = "/etc/fm/fm_srch_af_th.conf";
+#ifdef FM_LEGACY_PATCHLOADER
+const char *const CALIB_DATA_NAME = "/data/app/Riva_fm_cal";
+#else
 const char *const CALIB_DATA_NAME = "/data/misc/fm/Riva_fm_cal";
+#endif
 
 #define V4L2_CTRL_CLASS_USER  0x00980000
 #define V4L2_CID_BASE  (V4L2_CTRL_CLASS_USER | 0x900)
diff --git a/libfm_jni/Android.mk b/libfm_jni/Android.mk
index 901bdda..c221e71 100644
--- a/libfm_jni/Android.mk
+++ b/libfm_jni/Android.mk
@@ -24,6 +24,10 @@ LOCAL_SHARED_LIBRARIES := \
 #LOCAL_C_INCLUDES += $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
 #LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
 
+ifeq ($(TARGET_FM_LEGACY_PATCHLOADER),true)
+    LOCAL_CFLAGS += -DFM_LEGACY_PATCHLOADER
+endif
+
 LOCAL_MODULE := libfmjni
 include $(BUILD_SHARED_LIBRARY)
 
diff --git a/libfm_jni/FM_Const.h b/libfm_jni/FM_Const.h
index b0cc684..2f850d2 100644
--- a/libfm_jni/FM_Const.h
+++ b/libfm_jni/FM_Const.h
@@ -121,7 +121,11 @@ const char *const FM_INIT_PROP = "hw.fm.init";
 const char *const SCRIPT_START_PROP = "ctl.start";
 const char *const FM_SOC_DL_SCRIPT = "fm_dl";
 const char *const SCRIPT_STOP_PROP = "ctl.stOP";
+#ifdef FM_LEGACY_PATCHLOADER
+const char *const CALIB_DATA_NAME = "/data/app/Riva_fm_cal";
+#else
 const char *const CALIB_DATA_NAME = "/data/misc/fm/Riva_fm_cal";
+#endif
 const char *const SOC_PATCH_DL_SCRPT = "fm_dl";
 const char *const FM_DEVICE_PATH = "/dev/radio0";
 const char *const FM_PERFORMANCE_PARAMS = "/etc/fm/fm_srch_af_th.conf";
-- 
2.17.1

